esp-hal/src/ledc/timer.rs

```diff
@@ -179,11 +179,19 @@ pub trait TimerHW<S: TimerSpeed> {
 
     /// Update the timer in HW
     fn update_hw(&self);
+
+    /// Pause the timer
+    fn pause(&self);
+
+    /// Resume the timer
+    fn resume(&self);
+
+    /// Reset the timer
+    fn reset(&self);
 }
 
 /// Timer struct
-pub struct Timer<'a, S: TimerSpeed> {
-    ledc: &'a crate::peripherals::ledc::RegisterBlock,
+pub struct Timer<S: TimerSpeed> {
```

Comment:
I am not sure but I think we don't want the `Timer`s to potentially survive our use of `LEDC`. If the user takes a Timer, drops Ledc, reinitializes it and grabs the same Timer, we'll end up in a bad spot.

Reply:
Thank you for pointing this out. I have reintroduced the lifetime parameter for the timer.

Reply:
Thank you for pointing this out. I have reintroduced the lifetime parameter for the timer.